### PR TITLE
[RAPTOR-4520,4712] fix dataset temp filepath naming, release v1.4.15

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
1.
Fix temp dataset filepath generation during validation.
I've been using a column name in the filename. So in this bug, it happen that column's name was `smth/smth` :)

Replace filename with `null_value_imputation_columnXYZ`

2.
Fix a small regression I introduced, for the case of calling `drum` without any params.

3.
Added a fix for Py/R objects casting. I've been changing type of the `_class_labels` in-place, but this parameter is passed into 
`datarobot.mlops` API after predictions are made.

### CODE REVIEW

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

**Comment by yakov-g** [comment]
jarvis test functional_training_model_templates

File: custom_model_runner/datarobot_drum/drum/perf_testing.py [hunk 1]
```diff
@@ -523,25 +523,21 @@ def validation_test(self):
...
-            ) as temp_f:
-                temp_data_name = temp_f.name
-                df_tmp = df.copy()
-                df_tmp[column_name] = None
-                df_tmp.to_csv(temp_data_name, index=False)
-                CMRunnerUtils.replace_cmd_argument_value(
-                    cmd_list, ArgumentsOptions.INPUT, temp_data_name
-                )
+        for i, column_name in enumerate(column_names):
+            tmp_dataset_file_path = os.path.join(
```
Comment (rvorobii) [inline custom_model_runner/datarobot_drum/drum/perf_testing.py:527]: doesn't this need to be removed after it's used?

File: custom_model_runner/datarobot_drum/drum/perf_testing.py [hunk 2]
```diff
@@ -523,25 +523,21 @@ def validation_test(self):
...
+                null_datasets_dir, "null_value_imputation_column{}".format(i)
```
Comment (rvorobii) [inline custom_model_runner/datarobot_drum/drum/perf_testing.py:528]: how does this fixes the issue?

the file name still can be "null_value_imputation_columnsmth/smth" if column name is "smth/smth"

Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/perf_testing.py:528]: I'm not adding a column name into path anymore

File: custom_model_runner/datarobot_drum/drum/perf_testing.py (hunk 1 above)
Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/perf_testing.py:527]: Whole folder is removed if test passes and kept if test fails.

File: custom_model_runner/datarobot_drum/drum/perf_testing.py (hunk 2 above)
Comment (rvorobii) [reply custom_model_runner/datarobot_drum/drum/perf_testing.py:528]: oh, you're right, sorry

**Review by rvorobii — APPROVED** [review]

**Review by meltzg — CHANGES_REQUESTED** [review]
need a test testing that perf testing works with in `/` column names

**Comment by yakov-g** [comment]
Added a test case which fails without the path fix.

**Comment by yakov-g** [comment]
@rvorobii 
added more testing and one more fix for https://datarobot.atlassian.net/browse/RAPTOR-4712
I want them to go in ASAP.

**Review by meltzg — APPROVED** [review]

**Pull request merged**